pkgs/tools/nix/nixos-render-docs/README.md

```diff
@@ -0,0 +1,82 @@
+# nixos-render-docs
+
+A [CommonMark](https://commonmark.org/) and [`man-pages`](https://www.man7.org/linux/man-pages/man7/man-pages.7.html) renderer for the NixOS and Nixpkgs manuals.
+
+## Summary
+
+NixOS Render Docs is a renderer originally made for documentation written in the [DocBook](https://docbook.org/whatis) format, which was then dropped in support for CommonMark, a flavour of markdown, following [this RFC](https://github.com/NixOS/rfcs/pull/72). This shift aligns the project with more accessible documentation formats and community standards (along with the recent shift to [Diátaxis](https://diataxis.fr/)), enabling easier contributions and enhancing readability for new users.
+
+Having our own renderer gives us a few key benefits:
+- **Control and Flexibility**: We own the code, we can manipulate and extend it however we like. An example of this is the integration of option docs and our solution to redirecting away from broken links, as described in detail below.
+- **Scalability and Extensibility**: As the NixOS documentation grows, having a renderer under our control makes it easier to scale and introduce new features without being limited by the constraints of external tools. Future enhancements, such as tested code blocks or localized versions of documentation, can be seamlessly integrated.
+
+Several alternatives were considered, a detailed analysis on the trade-offs with each alternative can be found here: https://ethercalc.net/dc4vcnnl8zv0
+
+## Redirects system
+
+Moving contents around can cause links to break.
+
+Since we have our own markdown parser, we can hook into the rendering process to extract all of the metadata around each content identifier.
+The [mechanism for checking correctness of redirects](./src/nixos_render_docs/redirects.py) takes the collection of identifiers and a static mapping of redirects, validates them against a set of rules, and creates a client-side redirect mapping for each output path, as well as a `_redirects` file for server-side redirects in [Netlify syntax](https://docs.netlify.com/routing/redirects/#syntax-for-the-redirects-file).
+
+This allows us to catch:
+- Identifiers that were removed or renamed
+- Content that was moved from one location to another
+- Various consistency errors in the redirects mapping
+
+### Design considerations
+
+The creation, movement, and removal of every identifier is captured in the Git history of the Nixpkgs repo.
+However, leveraging this information during the build process of `nixos-render-docs` is quite tricky.
+There are two key inputs to the redirect system:
+
+1. The current set of identifiers
+2. A list of their historical locations
+
+One is accessible during the build process, and the other by fetching the Git history.
+Doing both of these during the build is not feasible, and to have an external command handling the Git history would require some form of communication between it and the build process.
+This quickly gets complicated.
```

Comment:
```suggestion
However, analysing hundreds of thousands of commits during the build process is impractical.
```

Let's not go into vagueness about alternatives considered. Either spell them out or leave them out.